icicle/include/icicle/fields/field.h

```diff
@@ -747,14 +369,37 @@ class Field
     return rs;
   }
 
-  template <unsigned MODULUS_MULTIPLE = 1>
   static constexpr HOST_DEVICE_INLINE Wide mul_wide(const Field& xs, const Field& ys)
   {
     Wide rs = {};
     multiply_raw(xs.limbs_storage, ys.limbs_storage, rs.limbs_storage);
     return rs;
   }
 
+#ifdef BARRET
+
+  static HOST_DEVICE_INLINE void
+  multiply_and_add_lsb_neg_modulus_raw(const ff_storage& as, ff_storage& cs, ff_storage& rs)
```

Comment:
I think you'd better move the logic to host_math and device_math like all the others to avoid this ifdef everywhere

Reply:
this is ifdef barret